internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -255,33 +257,34 @@ func (r *Reconciler) getMachineSetsForDeployment(ctx context.Context, d *cluster
 	// List all MachineSets to find those we own but that no longer match our selector.
 	machineSets := &clusterv1.MachineSetList{}
 	if err := r.Client.List(ctx, machineSets, client.InNamespace(d.Namespace)); err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to list MachineSets")
 	}
 
 	filtered := make([]*clusterv1.MachineSet, 0, len(machineSets.Items))
 	for idx := range machineSets.Items {
 		ms := &machineSets.Items[idx]
-		log.WithValues("MachineSet", klog.KObj(ms))
+		log = log.WithValues("MachineSet", klog.KObj(ms))
 		selector, err := metav1.LabelSelectorAsSelector(&d.Spec.Selector)
 		if err != nil {
-			log.Error(err, "Skipping MachineSet, failed to get label selector from spec selector")
+			log.Error(err, "Skipping MachineSet, failed to get label selector from spec selector", "MachineSet", klog.KObj(ms))
```

Comment:
```suggestion
			log.Error(err, "Skipping MachineSet, failed to get label selector from spec selector")
```
We don't have to add the MachineSet here. It's added in l.266 to the logger (at least after your fix :))

same in other cases below